recipes/armadillo/all/conanfile.py

```diff
@@ -0,0 +1,414 @@
+        self._cmake.definitions["ARMA_USE_ATLAS"] = (
+            self.options.use_lapack == "system_atlas"
+        )
+        self._cmake.definitions["ARMA_USE_HDF5"] = self.options.use_hdf5
+        self._cmake.definitions["ARMA_USE_ARPACK"] = self.options.use_arpack
+        self._cmake.definitions["ARMA_USE_EXTERN_RNG"] = self.options.use_extern_rng
+        self._cmake.definitions["ARMA_USE_SUPERLU"] = self.options.use_superlu
+        self._cmake.definitions["ARMA_USE_WRAPPER"] = self.options.use_wrapper
+        self._cmake.definitions["ARMA_USE_ACCELERATE"] = (
+            self.options.use_blas == "framework_accelerate"
+            or self.options.use_lapack == "framework_accelerate"
+        ) and self.settings.os == "Macos"
+        self._cmake.definitions["DETECT_HDF5"] = self.options.use_hdf5
+        self._cmake.definitions["USE_OPENBLAS"] = self.options.use_blas == "openblas"
+        self._cmake.definitions["USE_SYSTEM_LAPACK"] = (
+            self.options.use_lapack == "system_lapack"
+        )
+        self._cmake.definitions["USE_SYSTEM_BLAS"] = (
+            self.options.use_blas == "system_blas"
+        )
+        self._cmake.definitions["USE_SYSTEM_ATLAS"] = (
+            self.options.use_lapack == "system_atlas"
+        )
+        self._cmake.definitions["USE_SYSTEM_HDF5"] = (
+            self.options.use_hdf5 == "system_hdf5"
+        )
+        self._cmake.definitions["USE_SYSTEM_ARPACK"] = self.options.use_arpack
+        self._cmake.definitions["USE_SYSTEM_SUPERLU"] = self.options.use_superlu
+        self._cmake.definitions["USE_SYSTEM_OPENBLAS"] = (
+            self.options.use_blas == "system_openblas"
+        )
+        self._cmake.definitions["USE_SYSTEM_FLEXIBLAS"] = (
+            self.options.use_blas == "system_flexiblas"
+        )
+        self._cmake.definitions["USE_SYSTEM_MKL"] = (
+            self.options.use_blas == "system_mkl"
+        )
+        self._cmake.definitions["ALLOW_FLEXIBLAS_LINUX"] = (
+            self.options.use_blas == "system_flexiblas" and self.settings.os == "Linux"
+        )
+        self._cmake.definitions["ALLOW_OPENBLAS_MACOS"] = (
+            self.options.use_blas == "openblas"
+            or self.options.use_blas == "system_openblas"
+        ) and self.settings.os == "Macos"
+        self._cmake.definitions["ALLOW_BLAS_LAPACK_MACOS"] = (
+            self.options.use_blas != "framework_accelerate"
+        )
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder,
+            filename="{name}-{version}.tar.xz".format(
+                name=self.name, version=self.version
+            ),
+        )
+        tools.replace_in_file(
```

Comment:
Could these patches be exported to a patch file? Anyway, it should be applied as first step in `build()` method. Here an example: https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/jwasm/all/conanfile.py#L33

Reply:
Yep, if you think this is a better/cleaner way to approach it. This was one of the things I was unsure about, so I'll go ahead and make those changes.